app/models/tag.rb

```diff
@@ -36,6 +36,22 @@ class Tag < ActsAsTaggableOn::Tag
     searchableAttributes %w[name short_summary]
   end
 
+  def index_to_elasticsearch
+    Search::TagEsIndexWorker.perform_async(id)
+  end
+
+  def index_to_elasticsearch_inline
+    Search::Tag.index(id, serialized_search_hash)
+  end
+
+  def serialized_search_hash
+    Search::TagSerializer.new(self).serializable_hash.dig(:data, :attributes)
+  end
+
+  def elasticsearch_doc
+    Search::Tag.find_document(id)
+  end
```

Comment:
this raises an exception if called on an unindexed item, do we care?

```ruby
[14] pry(main)> tg.elasticsearch_doc
ETHON: performed EASY effective_url=http://localhost:9200/tags_development_alias/_doc/1 response_code=404 return_code=ok total_time=0.002645
2020-02-11 15:17:25 +0100: GET http://localhost:9200/tags_development_alias/_doc/1 [status:404, request:0.004s, query:N/A]
2020-02-11 15:17:25 +0100: < {"_index":"tags_development_alias","_type":"_doc","_id":"1","found":false}
2020-02-11 15:17:25 +0100: [404] {"_index":"tags_development_alias","_type":"_doc","_id":"1","found":false}
Elasticsearch::Transport::Transport::Errors::NotFound: [404] {"_index":"tags_development_alias","_type":"_doc","_id":"1","found":false}
from /Users/rhymes/.rvm/gems/ruby-2.6.5/gems/elasticsearch-transport-7.4.0/lib/elasticsearch/transport/transport/base.rb:205:in `__raise_transport_error'
```

Reply:
Great point! Back at Kenna right from the start we patched the request method of the ES gem to return an empty hash any time a 404 was raised. I don't know why the decision was originally made but down the road it led to some instances where we didn't fail as fast as we should have. This time around I didn't want to mess with the default until I had a good reason to.

Reply:
makes sense! I wouldn't encourage the patch, more like "rescue -> do something" :D but letting it blow up makes sense as well!